src/borg/testsuite/archiver.py

```diff
@@ -3742,60 +3742,76 @@ def get_changes(filename, data):
             joutput = [json.loads(line) for line in output.split('\n') if line]
 
             # File contents changed (deleted and replaced with a new file)
-            # assert 'file_replaced' in output  # added to debug #3494
-            assert any(('type', 'modified') in chg.items()
-                    for chg in get_changes('input/file_replaced', joutput))
+            if can_compare_ids:
+                assert {'type': 'modified', 'added': 4096, 'removed': 1024} in get_changes('input/file_replaced', joutput)
+            else:
+                assert {'type': 'modified'} in get_changes('input/file_replaced', joutput)
 
             # File unchanged
             assert not any(get_changes('input/file_unchanged', joutput))
 
             # Directory replaced with a regular file
             if 'BORG_TESTS_IGNORE_MODES' not in os.environ:
-                assert any((('type', 'mode') in chg.items()) and (('oldmode', 'drwxr-xr-x') in chg.items()) and
-                        (('newmode', '-rwxr-xr-x') in chg.items()) for chg in
-                        get_changes('input/dir_replaced_with_file', joutput))
+                assert {'type': 'mode', 'oldmode': 'drwxr-xr-x', 'newmode': '-rwxr-xr-x'} in \
+                    get_changes('input/dir_replaced_with_file', joutput)
 
             # Basic directory cases
-            assert any(('type', 'added directory') in chg.items() for chg in get_changes('input/dir_added', joutput))
-            assert any(('type', 'removed directory') in chg.items() for chg in get_changes('input/dir_removed', joutput))
+            assert {'type': 'added directory'} in get_changes('input/dir_added', joutput)
+            assert {'type': 'removed directory'} in get_changes('input/dir_removed', joutput)
 
             if are_symlinks_supported():
                 # Basic symlink cases
-                assert any(('type', 'changed link') in chg.items() for chg in get_changes('input/link_changed', joutput))
-                assert any(('type', 'added link') in chg.items() for chg in get_changes('input/link_added', joutput))
-                assert any(('type', 'removed link') in chg.items() for chg in get_changes('input/link_removed', joutput))
+                assert {'type': 'changed link'} in get_changes('input/link_changed', joutput)
+                assert {'type': 'added link'} in get_changes('input/link_added', joutput)
+                assert {'type': 'removed link'} in get_changes('input/link_removed', joutput)
 
                 # Symlink replacing or being replaced
-                assert any(('type', 'mode') in chg.items() for chg in get_changes('input/dir_replaced_with_link', joutput))
-                assert any(('type', 'mode') in chg.items() for chg in get_changes('input/link_replaced_by_file', joutput))
+                if sys.platform == "darwin":
+                    # wierd: on macOS, symlink mode is 'lrwx------'. Is this true, or is there an
+                    # underlying problem here.
```

Comment:
looks almost like the symlink permissions follow the umask on macOS, but not on Linux.

considering that this is the only difference, how about deduplicating this a bit and using:
```
expected_symlink_mode = 'lrwx------' if sys.platform == 'darwin' else 'lrwxrwxrwx
... # tests for both platforms in one
```

Reply:
could be also that on macOS it directly reflects the mode of the file it points to.